core/src/main/java/io/confluent/kafka/schemaregistry/rest/resources/RequestHeaderBuilder.java

```diff
@@ -17,19 +17,40 @@
 
 import org.eclipse.jetty.util.StringUtil;
 
+import javax.ws.rs.core.HttpHeaders;
 import java.util.HashMap;
+import java.util.List;
 import java.util.Map;
 
-import javax.ws.rs.core.HttpHeaders;
-
 public class RequestHeaderBuilder {
 
-  public Map<String, String> buildRequestHeaders(HttpHeaders httpHeaders) {
+  public Map<String, String> buildRequestHeaders(
+      HttpHeaders httpHeaders,
+      List<String> whitelistedHeaders
+  ) {
     Map<String, String> headerProperties = new HashMap<>();
+    addStaticHeaders(headerProperties, httpHeaders);
+    addWhitelistedHeaders(headerProperties, httpHeaders, whitelistedHeaders);
+    return headerProperties;
+  }
+
+  private void addWhitelistedHeaders(
+      Map<String, String> headerProperties,
+      HttpHeaders httpHeaders,
+      List<String> whitelistedHeaders
+  ) {
+    if (whitelistedHeaders == null) {
+      return;
+    }
+    whitelistedHeaders
```

Comment:
perhaps use `if (whitelistedHeaders != null)` and do streams().forEach if not null